libbeat/processors/add_host_geo/add_host_geo.go

```diff
@@ -0,0 +1,59 @@
+package add_geo
+
+import (
+	"fmt"
+
+	"github.com/elastic/beats/libbeat/beat"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/processors"
+)
+
+func init() {
+	processors.RegisterPlugin("add_host_geo", newAddGeo)
+}
+
+func newAddGeo(c *common.Config) (processors.Processor, error) {
+	config := &Config{}
+	c.Unpack(config)
+
+	ag := addGeo{
+		"name":             config.Name,
+		"location":         config.Location,
+		"continent_name":   config.ContinentName,
+		"country_iso_code": config.CountryISOCode,
+		"region_name":      config.RegionName,
+		"city_name":        config.CityName,
+	}
+	// Delete any empty values
+	for k, v := range ag {
+		if v == "" {
+			delete(ag, k)
+		}
+	}
+
+	return ag, nil
+}
+
+type Config struct {
+	Name           string `config:"name"`
+	Location       string `config:"location"`
+	ContinentName  string `config:"continent_name"`
+	CountryISOCode string `config:"country_iso_code"`
+	RegionName     string `config:"region_name"`
+	CityName       string `config:"city_name"`
+}
+
+type addGeo common.MapStr
+
+func (ag addGeo) Run(event *beat.Event) (*beat.Event, error) {
+	_, err := event.PutValue("host.geo", ag.asMap())
+	return event, err
+}
+
+func (ag addGeo) asMap() common.MapStr {
+	return common.MapStr(ag).Clone()
```

Comment:
nit: +1 on cloning, but from name only I would asMap expect to be a cast operation only. Caught me a little offguard when reading the argument for PutValue.

Reply:
Good point, I'll rename to `mapClone()`.